src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -65,6 +72,31 @@ public void NotifyLocationChanged(string uri, string state, bool intercepted)
         NotifyLocationChanged(intercepted);
     }
 
+    public void HandleLocationChanging(int callId, string uri, bool intercepted)
+    {
+        NotifyLocationChanging(uri, intercepted, result =>
+        {
+            bool success;
+
+            if (result.Exception is { } exception)
+            {
+                Log.NavigationFailed(_logger, uri, exception);
+                success = false;
+            }
+            else if (result.Canceled)
+            {
+                Log.NavigationCanceled(_logger, uri);
+                success = false;
+            }
+            else
+            {
+                success = true;
+            }
+
+            _jsRuntime.InvokeVoidAsync(Interop.EndLocationChanging, callId, success).Preserve();
+        });
+    }
```

Comment:
Why is this done this way vs how we do location changed in the server (see https://github.com/dotnet/aspnetcore/blob/main/src/Components/Server/src/ComponentHub.cs#L288-L297)

I believe this has implications with regards to things like how exceptions are dealt with when they happen at different stages.